### ISSUE TITLE
[Feature] Google Sheets API>New Method: Fetch Files("folderId")

### ISSUE THREAD

**Issue by GreenFlux** [appsmithorg/appsmith#4304]
## Summary

Requesting an additional method for Google Sheets API to fetch an array of all files _(and associated meta-data)_ for a given **folder Id**,  instead of _rows_ for a given **sheet Id**. The method should return an array of JSON objects representing all files in the given folder _(matching optional filter conditions)_.

## Motivation

Provide developers the ability to monitor a folder as a data-source, with each row of data being driven by the folder's current list of files and meta-data, including EXIF/Geo-location, dimensions, file size, file type, etc. 

### Use Cases:
- users bulk-upload photos to Google Folder => table populates with multiple rows representing "inspection_photo", "survey_location", "timestamp", "filename", etc
- User captures photo with auto-upload to Google folder already configured. Taking a photo will create a row that captures the user's location, along with the timestamp and photo as a row of data. 
- Other script/integration saves email attachments to a folder:  `if (Label="X") {saveAttachments ("folderId")}` => view as table in Appsmith
- Google form with file upload => Save files to folder and monitor as data-source

### Similar feature on AppSheet:
[https://help.appsheet.com/en/articles/4865398-folder-data-source-the-essentials](https://help.appsheet.com/en/articles/4865398-folder-data-source-the-essentials)

**Comment by Nikhil-Nandagopal** [comment]
@GreenFlux thank you for the feature request! This seems unrelated to Google Sheets however and I think a separate Google Drive Integration would be a better fit here?

**Comment by GreenFlux** [comment]
> @GreenFlux thank you for the feature request! This seems unrelated to Google Sheets however and I think a separate Google Drive Integration would be a better fit here?

Agreed. It is a separate Google API endpoint so it's not really a new method for the Sheets API. 

As a separate API, it could have a whole list of other methods like fetch files, create file, update file, delete file, etc.

**Comment by fabform** [comment]
Handy to [find google sheet id](https://fabform.io/a/how-to-find-google-sheet-id)

**Comment by mattbOSA** [comment]
Hello, just sharing our use case for this feature.  

We have all of our non-sensitive documents are hosted on google drive, and we are using appsmith to build an HR system to track our applicants path to an employee and our employees properties/stats (such as expiration of certifications, date of birth, progress reports, emergency contacts, etc).  

In order to more accurately track employee properties and statuses, I want to link every status/property of our employees to the supporting documentation that is hosted on google drive.  For example, if a certification is expiring on a certain date, I want to be able to click on that record and verify the expiration date by viewing the actual document (which is saved on google drive).  And this brings me back full circle to the details of our similar feature request.

We need a method of doing the following:
- Searching for, browsing for, and finally selecting a file on google drive and having that file's document's ID or document url returned to appsmith
- Uploading a file to google drive, optionally over-writing the file's original name, choosing where the file will be stored, and finally having that uploaded file's document ID or document URL returned to appsmith

Here is a short gif explaining our needs and use case.
https://drive.google.com/file/d/1adwd1ZpuxbZUNv6q8kyVZooJMph_s2M8/view?usp=sharing
